helix-core/src/main/java/org/apache/helix/monitoring/metrics/WagedRebalancerMetricCollector.java

```diff
@@ -38,7 +41,11 @@
 
     // The following latency metrics are related to AssignmentMetadataStore
     StateReadLatencyGauge,
-    StateWriteLatencyGauge
+    StateWriteLatencyGauge,
+
+    // Count of any rebalance failure.
+    // Note the rebalancer may still be able to return an assignment based on the previous record on an error.
+    RebalanceFailureCount
```

Comment:
I actually prefer making the name clearer. 

Pick one:
RebalancerFailureResourceCount? 
RebalancerFailurePartitionCount?

Reply:
It is logically the rebalancing failure count. Not related to resource or partition.
Actually, we don't plan to count failure partitions or resources. Because the rebalancer will stop on the first failed assignment attempt.